common/src/test/scala/uk/ac/wellcome/sqs/SQSReaderTest.scala

```diff
@@ -16,139 +17,149 @@ class SQSReaderTest
     extends FunSpec
     with Matchers
     with ScalaFutures
-    with SQSLocal {
+    with Eventually
+    with SqsFixtures {
 
-  val queueUrl = createQueueAndReturnUrl("test_queue")
-  // Setting 1 second timeout for tests, so that test don't have to wait too long to test message deletion
-  sqsClient.setQueueAttributes(queueUrl, Map("VisibilityTimeout" -> "1"))
+  override def withFixture(test: OneArgTest) = withLocalSqsQueue { queueUrl =>
```

Comment:
Can you talk us through the advantages of this pattern over the previous method of creating test resources?

Reply:
Basically, the `with*` fixture methods allow us to safely create and destroy stateful resources like SQS queues without relying on thread-unsafe techniques like using mutable variables. It also allows us to create resources with randomized names to improve test isolation. 

Lots of the tests create stateful stuff like SQS queues without tearing them down afterwards.

Reply:
You can read more about the Scalatest "official" techinques for sharing fixtures here:

http://www.scalatest.org/user_guide/sharing_fixtures

Reply:
Good enough for me.

Reply:
Oh really nice! So much better than before